backend/app/services/rag/retrieval_service.py

```
@@ -419,3 +419,93 @@ async def _retrieve_from_kb_internal(
+        Raises:
+            ValueError: If knowledge base not found or configuration invalid
+        """
+        from app.models.kind import Kind
+
+        # Get knowledge base directly without permission check
+        kb = (
+            db.query(Kind)
+            .filter(
+                Kind.id == knowledge_base_id,
+                Kind.kind == "KnowledgeBase",
+                Kind.is_active == True,
```

Ruff (0.14.10)

Avoid equality comparisons to `True`; use `Kind.is_active:` for truth checks

Replace with `Kind.is_active`

(E712)